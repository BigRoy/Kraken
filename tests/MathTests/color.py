import json

from kraken.core.maths import *


color0 = Color()
color0.r = 1.0;
print "color:" + str(color0)
color1 = Color(1.0, 0.0, 0.74, 0.5);
print "color:" + str(color1)
print "lerp:" + str(color0.linearInterpolate(color1, 0.5))

print "rand:" + str(Color.randomColor())

