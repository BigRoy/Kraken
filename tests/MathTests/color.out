color:Color(<RTVal:+1.0>,<RTVal:+0.0>,<RTVal:+0.0>,<RTVal:+0.0>)
color:Color(<RTVal:+1.0>,<RTVal:+0.0>,<RTVal:+0.0>,<RTVal:+0.0>)
Traceback (most recent call last):
  File "MathTests\color.py", line 11, in <module>
    print "lerp:" + str(color0.linearInterpolate(color1, 0.5))
  File "..\Python\kraken\core\maths\color.py", line 355, in linearInterpolate
    return Color(self._rtval.linearInterpolate('Color', ks.rtVal('Scalar', t)))
AttributeError: KL compile failed: no resolution for method Color.linearInterpolate(io Float32)

